airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/format_command.py

```diff
@@ -75,16 +75,53 @@ def get_help_message(self) -> str:
             message = f"{message}."
         return message
 
-    def get_dir_to_format(self, dagger_client: dagger.Client) -> dagger.Directory:
-        """Get the directory to format according to the file_filter.
+    async def get_dir_to_format(self, dagger_client) -> Directory:
+        """Get a directory with all the source code to format according to the file_filter.
+        We mount the the files to format in a git container and remove all gitignored files.
+        It ensures we're not formatting files that are gitignored.
 
         Args:
-            dagger_client (dagger.Client): The dagger client to use to get the directory
+            dagger_client (dagger.Client): The dagger client to use to get the directory.
 
         Returns:
-            dagger.Directory: The directory to format
+            Directory: The directory with the files to format that are not gitignored.
         """
-        return dagger_client.host().directory(self.LOCAL_REPO_PATH, include=self.file_filter, exclude=DEFAULT_FORMAT_IGNORE_LIST)
+        # Load a directory from the host with all the files to format according to the file_filter and the .gitignore files
+        dir_to_format = dagger_client.host().directory(
+            self.LOCAL_REPO_PATH, include=self.file_filter + ["**/.gitignore", ".gitignore"], exclude=DEFAULT_FORMAT_IGNORE_LIST
+        )
+
+        return (
+            dagger_client.container()
+            .from_("alpine:latest")
+            .with_exec(
+                sh_dash_c(
+                    [
+                        "apk update",
+                        "apk add git",
+                    ]
+                )
+            )
+            .with_workdir(REPO_MOUNT_PATH)
+            .with_mounted_directory(REPO_MOUNT_PATH, dir_to_format)
+            # All with_exec commands below will re-run if the to_format directory changes
+            .with_exec(["git", "init"])
+            # This is needed to avoid git complaining about the user not being set
+            .with_exec(["git", "config", "user.email", "ci@airbyte.io"])
+            .with_exec(["git", "config", "user.name", "Airbyte CI"])
+            # Add all files to the git index
+            .with_exec(["git", "add", "."])
+            # Commit all files
+            .with_exec(["git", "commit", "--message", "Commit for formatting"])
```

Comment:
I don't believe that you need to commit anything for `git clean` to do its thing, in which case you can remove everything in between `init` and `clean`.

Reply:
If I don't commit the files they will be considered as untracked files - hence the commit

Reply:
It's true that they will be considered as untracked files, however, that does not affect how `clean` behaves. See for yourself, try it out.

Reply:
I'm also surprised that it works, mind you, so I looked into it. The `-X` flag for `clean` is the reason it works: https://git-scm.com/docs/git-clean